### PR TITLE
Allow to pass children to Filter & Follow button

### PR DESCRIPTION
**Pull request by edellucien**

### CODE REVIEW

File: src/components/FilterButton/FilterButton.js [hunk 1]
```diff
@@ -22,29 +22,37 @@ const propTypes = {
...
 
+  /**
+   * Button is active.
+   */
+  active: PropTypes.bool.isRequired,
+
+  /**
+   * Set active value.
+   */
+  setActive: PropTypes.func.isRequired,
```
Comment (oterral) [inline src/components/FilterButton/FilterButton.js]: wierd name you already have an active property.  `onChange` seems more appropriate

Comment (edellucien) [reply src/components/FilterButton/FilterButton.js]: Indeed onChange is better :)

File: src/components/FilterButton/FilterButton.js [hunk 2]
```diff
@@ -77,28 +84,39 @@ class FilterButton extends PureComponent {
...
+          filterFc = TrajservLayer.createFilter(
+            lineParam ? lineParam.split(',') : undefined,
+            routeParam ? routeParam.split(',') : undefined,
+            opParam ? opParam.split(',') : undefined,
+            regexPublishedLineName,
+            false,
+          );
+        }
+
+        trackerLayer.setFilter(filterFc);
```
Comment (oterral) [inline src/components/FilterButton/FilterButton.js]: seems a lot similar to addTrackerFilters function, is there a possibility to reuse it.

File: src/components/FollowButton/FollowButton.js [hunk 3]
```diff
@@ -21,6 +21,16 @@ const propTypes = {
...
 
+  /**
+   * Button is active.
+   */
+  active: PropTypes.bool.isRequired,
+
+  /**
+   * Function triggered on button change.
+   */
+  onChange: PropTypes.func.isRequired,
```
Comment (oterral) [inline src/components/FollowButton/FollowButton.js]: In fact, it's just a button. The function should be called onClick and then in the callback you change the value of active .

```
<FollowButton
  active={active}
  onClick={()=> {
     active=!active  
  });
/>
```

File: src/layers/TrajservLayer.js [hunk 4]
```diff
@@ -145,23 +141,23 @@ class TrajservLayer extends TrackerLayer {
    * @param {string} regexLine
    * @private
    */
-  static createFilter(line, trip, operator, regexLine) {
+  static createFilter(line, trip, operator, regexLine, isConstructor) {
```
Comment (oterral) [inline src/layers/TrajservLayer.js]: What's the point of isConstrcutor ?  It seems filterFc created in the constructor is overwritten in init() function by the `addTrackerFilters`.

**Comment by edellucien** [comment]
I left it only in start of the tracker layer. It does make more sense and is easier. Thanks :)

File: src/layers/TrajservLayer.js [hunk 5]
```diff
@@ -229,40 +219,49 @@ class TrajservLayer extends TrackerLayer {
...
       );
+    } else {
+      this.filterFc = null;
     }
 
-    if (this.tracker && this.filterFc) {
+    if (this.tracker) {
       this.tracker.setFilter(this.filterFc);
     }
-
```
Comment (oterral) [inline src/layers/TrajservLayer.js]: there is a `this.setFilter(this.filterFc)` that does that

File: src/layers/TrajservLayer.js [hunk 6]
```diff
@@ -229,40 +219,49 @@ class TrajservLayer extends TrackerLayer {
       return;
     }
 
+    // Sort the trajectories.
+    if (this.tracker && this.sortFc) {
+      this.tracker.setSort(this.sortFc);
```
Comment (oterral) [inline src/layers/TrajservLayer.js]: Like `this.setFIlter` we should have a `this.setSort` in TrackerLayer

**Comment by edellucien** [comment]
This is intended to be a filter on the features, but not to select features.

File: src/layers/TrackerLayer.js [hunk 7]
```diff
@@ -210,6 +210,16 @@ class TrackerLayer extends Layer {
     }
   }
 
+  /**
+   * Set the sort for tracker features.
+   * @param {Function} sort Sort function.
+   */
+  setSort(sort) {
+    if (this.tracker && sort) {
```
Comment (oterral) [inline src/layers/TrackerLayer.js]:
```suggestion
    if (this.tracker) {
```
We want to be able to remove the sort if necessary

**Pull request merged**